pydoro/pydoro_core/config.py

```diff
@@ -106,6 +113,14 @@ def _ini_load(self):
         self.long_break_minutes = float(self._conf["Time"]["long_break_minutes"])
         self.alarm_seconds = int(self._conf["Time"]["alarm_seconds"])
         self.key_bindings = self._conf["KeyBindings"]
+        self.work_state_cmd = \
```

Comment:
This is not backwards compatible with what we already got in .ini. Perhaps put this section in a try catch

and if there's a KeyError set the values to empty arrays `[]` 👍🏽

Reply:
```python
try:
	self.work_state_cmd = \        
		ast.literal_eval(self._conf["Trigger"]["work_state_cmd"])
	self.work_paused_state_cmd = \
		ast.literal_eval(self._conf["Trigger"]["work_paused_state_cmd"])
	self.small_break_state_cmd = \
		ast.literal_eval(self._conf["Trigger"]["small_break_state_cmd"])
	self.long_break_state_cmd = \
		ast.literal_eval(self._conf["Trigger"]["long_break_state_cmd"])
except KeyErorr:
	self.work_state_cmd = []
	self.work_paused_state_cmd = []
	self.small_break_state_cmd = []
	self.long_break_state_cmd = []
```

Reply:
> ```python
> try:
> 	self.work_state_cmd = \        
> 		ast.literal_eval(self._conf["Trigger"]["work_state_cmd"])
> 	self.work_paused_state_cmd = \
> 		ast.literal_eval(self._conf["Trigger"]["work_paused_state_cmd"])
> 	self.small_break_state_cmd = \
> 		ast.literal_eval(self._conf["Trigger"]["small_break_state_cmd"])
> 	self.long_break_state_cmd = \
> 		ast.literal_eval(self._conf["Trigger"]["long_break_state_cmd"])
> except KeyError:
> 	self.work_state_cmd = []
> 	self.work_paused_state_cmd = []
> 	self.small_break_state_cmd = []
> 	self.long_break_state_cmd = []
> ```

I Think setting the default value of the config and updating them when a key is present in `.ini` file is a better solution.  This will also help if a line gets accidentally deleted in the config, and the config file can be short and user-specific.

Reply:
Hmm it doesn't work the way you think it does. 

Default values are written to config file if the file does not exist.

If some user already has pydoro and updates to a new version, it will not be able to find `Trigger` section as it was not there in previous versions of pydoro, when it wrote the `.ini` file.

Hope that make sense.

Reply:
Alternatively we can do it like this.

```python

if not self._conf["Trigger"]:
	self._conf["Trigger"] = {}
	self._conf["Trigger"]["work_state_cmd"] = "[]"
	self._conf["Trigger"]["work_paused_state_cmd"] = "[]"
	self._conf["Trigger"]["long_break_state_cmd"] = "[]"
	self._conf["Trigger"]["small_break_state_cmd"] = "[]"

self.work_state_cmd = \
	ast.literal_eval(self._conf["Trigger"]["work_state_cmd"])
self.work_paused_state_cmd = \
	ast.literal_eval(self._conf["Trigger"]["work_paused_state_cmd"])
self.small_break_state_cmd = \
	ast.literal_eval(self._conf["Trigger"]["small_break_state_cmd"])
self.long_break_state_cmd = \
	ast.literal_eval(self._conf["Trigger"]["long_break_state_cmd"])
```

Reply:
Check out c4a13f8, with this the behaviour remains same, if no config file is present, but partial config files are also supported.

Reply:
Now  The following is a valid config file:
```
[Trigger]
work_state_cmd = ["notify-send", "work_state_cmd"]
work_paused_state_cmd = ["notify-send", "work_paused_state_cmd"]
long_break_state_cmd = ["notify-send", "long_break_state_cmd"]
small_break_state_cmd = ["notify-send", "small_break_minutes"]
```
this will also prevent any backwards compatibility down the line.